code/modules/games/cards.dm

```diff
@@ -55,8 +59,16 @@
 /obj/item/weapon/deck/attackby(obj/O as obj, mob/user as mob)
 	if(istype(O,/obj/item/weapon/hand))
 		var/obj/item/weapon/hand/H = O
-		if(H.parentdeck == src)
+		if(decklimit && (decklimit <= cards.len))
+			to_chat(user,"<span class='warning'>This deck is full!</span>")
+			return
+		if(H.parentdeck == src.decktype)
```

Comment:
See above note about using typepaths here, this would become an `istype(playing card, decktype)`